eva-accession-release/src/test/java/uk/ac/ebi/eva/accession/release/configuration/CreateReleaseStepConfigurationTest.java

```diff
@@ -0,0 +1,146 @@
+
+import com.lordofthejars.nosqlunit.annotation.UsingDataSet;
+import com.lordofthejars.nosqlunit.core.LoadStrategyEnum;
+import com.lordofthejars.nosqlunit.mongodb.MongoDbConfigurationBuilder;
+import com.lordofthejars.nosqlunit.mongodb.MongoDbRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.batch.core.BatchStatus;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.test.JobLauncherTestUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import uk.ac.ebi.eva.accession.release.parameters.InputParameters;
+import uk.ac.ebi.eva.accession.release.test.configuration.BatchTestConfiguration;
+import uk.ac.ebi.eva.accession.release.test.configuration.MongoTestConfiguration;
+import uk.ac.ebi.eva.accession.release.test.rule.FixSpringMongoDbRule;
+import uk.ac.ebi.eva.commons.core.utils.FileUtils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static uk.ac.ebi.eva.accession.release.configuration.BeanNames.CREATE_RELEASE_STEP;
+
+@RunWith(SpringRunner.class)
+@ContextConfiguration(classes = {BatchTestConfiguration.class, MongoTestConfiguration.class})
+@UsingDataSet(locations = {
+        "/test-data/dbsnpClusteredVariantEntity.json",
+        "/test-data/dbsnpSubmittedVariantEntity.json"})
+@TestPropertySource("classpath:application.properties")
+public class CreateReleaseStepConfigurationTest {
+
+    private static final String TEST_DB = "test-db";
+
+    private static final long EXPECTED_LINES = 3;
+
+    @Autowired
+    private JobLauncherTestUtils jobLauncherTestUtils;
+
+    @Autowired
+    private InputParameters inputParameters;
+
+    //Required by nosql-unit
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Rule
+    public MongoDbRule mongoDbRule = new FixSpringMongoDbRule(
+            MongoDbConfigurationBuilder.mongoDb().databaseName(TEST_DB).build());
+
+    @Test
+    public void contextLoads() {
+
+    }
+
+    @Test
+    public void basicJobCompletion() throws Exception {
+        JobExecution jobExecution = jobLauncherTestUtils.launchStep(CREATE_RELEASE_STEP);
+        assertEquals(BatchStatus.COMPLETED, jobExecution.getStatus());
+    }
+
+    @Test
+    public void variantsWritten() throws Exception {
+        JobExecution jobExecution = jobLauncherTestUtils.launchStep(CREATE_RELEASE_STEP);
+        assertEquals(BatchStatus.COMPLETED, jobExecution.getStatus());
```

Comment:
Extract these two statements to share with basicJobCompletion()?

Reply:
they are not the same. the step test uses `launchStep` and the job one uses `launchJob`. I  extracted the lines and reused 4 times in the test, although I think it's quite clear as it is.